src/pages/_app.js

```diff
@@ -212,7 +216,21 @@ TreetrackerApp.getInitialProps = async (context) => {
 
   const device = userAgentFromString(userAgent)?.device.type || 'desktop';
 
-  return { props, device };
+  const mapConfigRequest = await fetch(
+    // TODO: use the ENV var, currently results in a bug with the theme editor
+    // `${process.env.NEXT_PUBLIC_CONFIG_API}/config`,
+    `https://dev-k8s.treetracker.org/map_config/config`,
+  );
+  const mapConfig = await mapConfigRequest.json();
+  const config =
+    mapConfig.find((item) => item.name === 'testing-config')?.data ||
+    defaultConfig;
+
+  return {
+    props,
+    config,
+    device,
+  };
```

Comment:
Do you think React context is a better way, so:
1. We don't need to pass through the config
2. The config is an infrequent change variable (99% of cases), so context fits this case.

Aslo, because we are in `beta` rather a dedicated branch, maybe we need a switcher (like env var) to open/close this feature, I mean, loading global config for the webmap, so we can close this on prod at this stage

Reply:
Yes you're right, using context is a better solution. I'll update the functionality to only run on DEVELOPMENT

Reply:
If you can find someone who can help us on this devops job then we will not be worried about this kind of thin. https://github.com/Greenstand/treetracker-web-map-client/issues/1071